src/LiveSplit.Core/Model/RunMetadata.cs

```diff
@@ -336,3 +381,63 @@ private void TriggerPropertyChanged(bool clearRunID)
         PropertyChanged?.Invoke(this, new MetadataChangedEventArgs(clearRunID));
     }
 }
+
+/// <summary>
+///     A custom variable that has a value and can be marked permanent.
+/// </summary>
+public sealed class CustomVariable
+{
+    /// <summary>
+    ///     The current value of the custom variable.
+    /// </summary>
+    public string Value { get; set; }
+
+    /// <summary>
+    ///     Gets a value indicating whether the custom variable should be saved in the run permanently.
+    /// </summary>
+    /// <value>
+    ///     <see langword="false"/> initially. Once set to <see langword="true"/>, will remain as such.
+    /// </value>
+    public bool IsPermanent { get; private set; }
+
+    /// <summary>
+    ///     Constructs an empty temporary custom variable.
```

Comment:
```suggestion
    ///     Creates a new instance of the <see cref="CustomVariable"/> class
    ///     with a default value and temporary permanence.
```

Not big on the phrasing "temporary permanence", maybe you can think of something better.